pkg/containers/containers.go

```diff
@@ -381,18 +381,17 @@ func (c *Containers) GetCgroupInfo(cgroupId uint64) CgroupInfo {
 		// getCgroupPath() to walk all cgroupfs directories until it finds the
 		// directory of given cgroupId.
 		c.cgroupsMutex.Lock()
-		defer c.cgroupsMutex.Unlock()
-
 		path, err := cgroup.GetCgroupPath(c.cgroups.GetDefaultCgroup().GetMountPoint(), cgroupId, "")
 		if err == nil {
 			var stat syscall.Stat_t
 			if err = syscall.Stat(path, &stat); err == nil {
-				info, err := c.cgroupUpdate(cgroupId, path, time.Unix(stat.Ctim.Sec, stat.Ctim.Nsec))
-				if err == nil {
-					return info
+				_, err := c.cgroupUpdate(cgroupId, path, time.Unix(stat.Ctim.Sec, stat.Ctim.Nsec))
```

Comment:
...doing this
```suggestion
				info, err = c.cgroupUpdate(cgroupId, path, time.Unix(stat.Ctim.Sec, stat.Ctim.Nsec))
```

Of course, that would require `var` declaring before.
This would save us from RLocking and RUnlocking after this path.